### PR TITLE
Clean up model index accessing

### PR DESCRIPTION
**Pull request by blink1073**

### CODE REVIEW

File: src/browser.ts [hunk 1]
```diff
@@ -167,7 +167,7 @@ class FileBrowserWidget extends Widget {
   open(): void {
     let foundDir = false;
     for (let index of this._model.selected) {
-      let item = this._model.items[index];
+      let item = this._model.getSortedItems()[index];
```
Comment (sccolbert) [inline src/browser.ts]: the call to `getSortedIttems()` should be hoisted out of the loop.

File: src/crumbs.ts [hunk 2]
```diff
@@ -233,7 +233,7 @@ class BreadCrumbs extends Widget {
     // Move all of the items.
     let promises: Promise<void>[] = [];
     for (let index of this._model.selected) {
-      var original = this._model.items[index].name;
+      var original = this._model.getSortedItems()[index].name;
```
Comment (sccolbert) [inline src/crumbs.ts]: hoist `getSortedItems()` out of the loop.

File: src/listing.ts [hunk 3]
```diff
@@ -783,7 +786,7 @@ class DirListing extends Widget {
       if (node.classList.contains(ITEM_CLASS)) {
         // Open the selected item.
         let index = this._items.indexOf(node);
-        let item = this._model.items[index];
+        let item = this._model.getSortedItems()[index];
```
Comment (sccolbert) [inline src/listing.ts]: hoist outside of loop.

**Comment by sccolbert** [comment]
Are the `selectedIndices` going to be out-of-sync now that widgets are operating on sorted copies? The model itself doesn't seem to use the selected indices. Should those be moved to the place where they are actually used?

**Comment by blink1073** [comment]
`model.selected` is used by three of the widgets.  I cleared it after a sort request to prevent stale selection.  Other comments addressed.

**Pull request merged**